Tools/clinic/clinic.py

```diff
@@ -1415,6 +1419,7 @@ def render_option_group_parsing(self, f, template_dict):
                     group = required
                 else:
                     right.append(group)
+            assert group is not None
```

Comment:
This is fine for now, but I feel like there's probably a simpler way of writing the code immediately above that would be easier to understand and not require this assertion here for mypy

Reply:
Yes. We should accumulate these smells in the modernising-issue, so we don't forget. Or we can just grep for asserts.